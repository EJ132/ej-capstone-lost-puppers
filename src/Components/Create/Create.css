div > .createPup {
    min-height: 120vh;
    height: auto;
}

.createPup {
    width: 100%;
    background: url("../../Images/create.jpg")
    no-repeat
    center
    center;
    background-size: 2000px 1100px;
    height: 95vh;
    position: relative;
    text-align: center;
}

.createPup div label, .createPup div input{
    margin-left: auto;
    margin-right: auto;
    text-align: center;
}

.createPup div label {
    width: 60%;
    margin-bottom: 10px;
    margin-top: 10px;
    height: 15px;
}

.createPup div input {
    height: 25px;
    width: 50%;
}

.createPup div select {
    width: 50%;
    height: 25px;
    margin-left: auto;
    margin-right: auto;
}

.createPup div button {
    width: 25%;
    height: 30px;
    border: none;
    border-radius: 5px;
    background-color: rgb(187, 177, 235);
    color: rgb(255, 253, 253);
    margin-left: auto;
    margin-right: auto;
    margin-top: 20px;
}

.createPup div {
    display: flex;
    flex-direction: column;
    justify-content: center;
    text-align: center;
    position: absolute;
    top: 250px;
    right: 10%;
    width: 500px;
}

.createPup div h2 {
    font-size: 3rem;
}

#submitted {
    font-size: 2rem;
    margin-top: 10px;
    color: green;
}

@media screen and (max-width: 1350px){
    .createPup div {
        right: 5%
    }
}

@media screen and (max-width: 768px) {
    .createPup {
        width: 100%;
        background: url("../../Images/create.jpg")
        no-repeat
        center
        center;
        background-size: cover;
        height: 95vh;
        position: relative;
    }
    
    .createPup div label, .createPup div input{
        margin-left: auto;
        margin-right: auto;
        text-align: center;
    }

    #imgLabel {
        height: 40px;
    }

    #descInput {
        height: 40px;
        border-radius: 5px;
        border: none;
    }
    
    .createPup div label {
        width: 60%;
        margin-bottom: 10px;
        margin-top: 10px;
        height: 20px;
    }
    
    .createPup div input {
        height: 25px;
        width: 50%;
        border-radius: 5px;
        border: none;
    }
    
    .createPup div select {
        width: 50%;
        height: 25px;
        margin-left: auto;
        margin-right: auto;
        justify-content: center;
        text-align: center;
        background-color: rgb(109, 109, 109);
        border-radius: 5px;
        color: white;
    }
    
    .createPup div {
        display: flex;
        flex-direction: column;
        justify-content: center;
        text-align: center;
        position: initial;
        padding-top: 40px;
        width: 100%;
    }
    
    .createPup div h2 {
        font-size: 3rem;
        margin-bottom: 120px;
    }
}