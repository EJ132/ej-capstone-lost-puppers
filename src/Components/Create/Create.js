import React, { Component } from 'react'
import NavBar from '../NavBar/Navbar'
import './Create.css'
import TokenService from '../../services/token-service'
import AuthApiService from '../../services/auth-api-service'
import history from '../../Context/history'

export default class Create extends Component {

    state = {
        error: null
    }

    createPup = ev => {
        ev.preventDefault()

        const {name, image, category, description, zipcode} = ev.target
        console.log(name.value, image.value, category.value, description.value, zipcode.value)

        const owner = TokenService.getUserId()

        this.setState({
            error: null
        })

        AuthApiService.postPup({
            name: name.value,
            image: image.value,
            category: category.value,
            description: description.value,
            zipcode: zipcode.value,
            owner: owner
        })
            .then(pup => {
                name.value = ''
                image.value = ''
                category.value = ''
                description.value = ''
                zipcode.value = ''
            })
            .then(() => history.push('/find'))
            .catch(res => {this.setState({error: res.error})})

        console.log(owner)
    }


    render(){

        const {error} = this.state;

        return(
            <div>

                <NavBar />

                <form className="createPup" onSubmit={this.createPup}>
                    <section className="regError" role='alert'>
                        {error && <p className='red'>{error}</p>}
                    </section>
                    <div>
                        <h2>Create A New Listing</h2>
                        <label>Name</label>
                        <input type='text' name='name' placeholder='Max' required></input>
                        <label id="imgLabel">Img (will support img uploading soon)</label>
                        <input type='text' name='image' placeholder='https://images.dog.ceo/breeds/mix/cheyenne1.jpg' required></input>
                        <label>Zipcode</label>
                        <input type='number' name='zipcode' placeholder='90250' required></input>
                        <label>Description</label>
                        <input id="descInput" placeholder='Please be as descriptive as possible' name='description' type='text' required></input>
                        <label>Category</label>
                        <select name='category' required>
                            <option value="Small">- Small -</option>
                            <option value="Medium">- Medium -</option>
                            <option value="Large">- Large -</option>
                        </select>
                        <button type="submit">Submit</button>
                    </div>
                </form>

                <footer>&#169; EJ Gonzalez 2019</footer>
            </div> 
        )
    }
}

//implementing city feature in future
// <label>City</label>
// <input type='text' name='city' placeholder='Hawthorne' required></input>