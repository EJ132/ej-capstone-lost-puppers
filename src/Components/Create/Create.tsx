import * as React from "react";

import history from "../../Context/history";
import AuthApiService from "../../services/auth-api-service";
import TokenService from "../../services/token-service";
import NavBar from "../NavigationBar/NavigationBar";

import "./Create.css";

export default function Create() {
  const [error, setError] = React.useState(null);
  const [selectedFile, setSelectedFile] = React.useState(null);
  const [submitted, setSubmitted] = React.useState(null);

  // ALL INFO IS APPENDED INTO A FORM DATA TO BE SENT AS FORM DATA (FILE UPLOADING)

  const createPup = (ev) => {
    ev.preventDefault();

    const { name, image, category, description, zipcode, lat, long } =
      ev.target;

    const owner = TokenService.getUserId();

    const formData = new FormData();
    formData.append("image", selectedFile, selectedFile.name);
    formData.append("name", name.value);
    formData.append("category", category.value);
    formData.append("description", description.value);
    formData.append("zipcode", zipcode.value);
    formData.append("lat", lat.value);
    formData.append("long", long.value);
    formData.append("owner", owner);

    setError(null);
    setSubmitted(true);

    AuthApiService.postPup(formData)
      .then((pup) => {
        name.value = "";
        image.value = "";
        category.value = "";
        description.value = "";
        zipcode.value = "";
        lat.value = "";
        long.value = "";
      })
      .then(() => history.push("/find"))
      .catch((res) => {
        setError(res.error);
      });
  };

  const fileUpload = (e) => {
    setSelectedFile(e.target.files[0]);
  };

  return (
    <div>
      <NavBar />

      <form
        className="createPup"
        encType="multipart/form-data"
        onSubmit={createPup}
      >
        <section className="regError" role="alert">
          {error && <p className="red">{error}</p>}
        </section>
        <div>
          <h2>Create A New Listing</h2>
          <label htmlFor="name">Name</label>
          <input type="text" name="name" placeholder="Max" required />
          <label id="imgLabel" htmlFor="image">
            Image
          </label>
          <input
            type="file"
            name="image"
            id="image"
            onChange={(e) => fileUpload(e)}
            required
          />
          <label htmlFor="zipcode">Zipcode</label>
          <input type="number" name="zipcode" placeholder="90250" required />
          <label htmlFor="lat">Lat (Get cordinates on google)</label>
          <input
            type="float"
            step="0.0000001"
            name="lat"
            placeholder="33.8877101"
            required
          />
          <label htmlFor="long">Long</label>
          <input
            type="float"
            step="0.0000001"
            name="long"
            placeholder="-118.3652527"
            required
          />
          <label htmlFor="description">Description</label>
          <input
            id="descInput"
            placeholder="Please be as descriptive as possible"
            name="description"
            type="text"
            required
          />
          <label htmlFor="category">Category</label>
          <select name="category" required>
            <option value="Small">- Small -</option>
            <option value="Medium">- Medium -</option>
            <option value="Large">- Large -</option>
          </select>
          <button type="submit">Submit</button>
          {submitted ? <p id="submitted">Submitted</p> : null}
        </div>
      </form>

      <footer>&#169; EJ Gonzalez 2019</footer>
    </div>
  );
}
