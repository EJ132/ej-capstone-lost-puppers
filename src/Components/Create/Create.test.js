import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Create from './Create'

describe(`Create component`, () => {

  it('renders a Create page by default', () => {
    const wrapper = shallow(<Create />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders only the form', () => {
      const form = shallow(<Create />)
        .find('.createPup')
      expect(toJson(form)).toMatchSnapshot()
  })
})
