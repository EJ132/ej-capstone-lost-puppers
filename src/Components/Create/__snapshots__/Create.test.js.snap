// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create component renders a Create page by default 1`] = `
<div>
  <Navbar />
  <form
    className="createPup"
    onSubmit={[Function]}
  >
    <section
      className="regError"
      role="alert"
    />
    <div>
      <h2>
        Create A New Listing
      </h2>
      <label>
        Name
      </label>
      <input
        name="name"
        placeholder="Max"
        required={true}
        type="text"
      />
      <label
        id="imgLabel"
      >
        Img (will support img uploading soon)
      </label>
      <input
        name="image"
        placeholder="https://images.dog.ceo/breeds/mix/cheyenne1.jpg"
        required={true}
        type="text"
      />
      <label>
        Zipcode
      </label>
      <input
        name="zipcode"
        placeholder="90250"
        required={true}
        type="number"
      />
      <label>
        Description
      </label>
      <input
        id="descInput"
        name="description"
        placeholder="Please be as descriptive as possible"
        required={true}
        type="text"
      />
      <label>
        Category
      </label>
      <select
        name="category"
        required={true}
      >
        <option
          value="Small"
        >
          - Small -
        </option>
        <option
          value="Medium"
        >
          - Medium -
        </option>
        <option
          value="Large"
        >
          - Large -
        </option>
      </select>
      <button
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
  <footer>
    Â© EJ Gonzalez 2019
  </footer>
</div>
`;

exports[`Create component renders only the form 1`] = `
<form
  className="createPup"
  onSubmit={[Function]}
>
  <section
    className="regError"
    role="alert"
  />
  <div>
    <h2>
      Create A New Listing
    </h2>
    <label>
      Name
    </label>
    <input
      name="name"
      placeholder="Max"
      required={true}
      type="text"
    />
    <label
      id="imgLabel"
    >
      Img (will support img uploading soon)
    </label>
    <input
      name="image"
      placeholder="https://images.dog.ceo/breeds/mix/cheyenne1.jpg"
      required={true}
      type="text"
    />
    <label>
      Zipcode
    </label>
    <input
      name="zipcode"
      placeholder="90250"
      required={true}
      type="number"
    />
    <label>
      Description
    </label>
    <input
      id="descInput"
      name="description"
      placeholder="Please be as descriptive as possible"
      required={true}
      type="text"
    />
    <label>
      Category
    </label>
    <select
      name="category"
      required={true}
    >
      <option
        value="Small"
      >
        - Small -
      </option>
      <option
        value="Medium"
      >
        - Medium -
      </option>
      <option
        value="Large"
      >
        - Large -
      </option>
    </select>
    <button
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
